src/routes/(app)/settings/profile/+page.svelte

```diff
@@ -23,24 +37,197 @@ $effect(() => {
+    for (const part of bodyPartsBytes) {
+        bodyBytes.set(part, offset);
+        offset += part.length;
+    }
+    bodyBytes.set(fileBytes, offset);
+    offset += fileBytes.length;
+    bodyBytes.set(encoder.encode(`\r\n--${boundary}--\r\n`), offset);
+
+    // Hash the complete body
+    const hashBuffer = await crypto.subtle.digest("SHA-256", bodyBytes);
+    const hashArray = Array.from(new Uint8Array(hashBuffer));
+    const hashHex = hashArray.map((b) => b.toString(16).padStart(2, "0")).join("");
+
+    // Make sure the URL is properly formatted
+    const fullUrl = new URL(apiUrl);
+
+    // Generate NIP-98 authorization token with the complete body hash
+    const authEvent = await invoke<string>("generate_nip98_auth_token", {
+        url: fullUrl.toString(),
+        method: "POST",
+        payload: hashHex,
+    });
+
+    // Base64 encode the auth event
+    const authEventBase64 = btoa(authEvent);
+
+    // Upload the file using the exact same body we hashed
+    const uploadResponse = await fetch(fullUrl.toString(), {
+        method: "POST",
+        headers: {
+            Authorization: `Nostr ${authEventBase64}`,
+            Accept: "application/json",
+            "Content-Type": `multipart/form-data; boundary=${boundary}`,
+        },
+        body: bodyBytes,
+    });
+
+    if (!uploadResponse.ok) {
+        const errorText = await uploadResponse.text();
+        console.error("Upload failed with status:", uploadResponse.status);
+        console.error("Error response:", errorText);
+        console.error("Auth token used:", authEventBase64);
+        throw new Error(`Failed to upload image (${uploadResponse.status}): ${errorText}`);
+    }
+
+    const result = await uploadResponse.json();
+    if (!result.nip94_event?.tags) {
+        throw new Error("Invalid response from server");
+    }
+
+    // Find the URL tag in the response
+    const urlTag = result.nip94_event.tags.find((tag: string[]) => tag[0] === "url");
+    if (!urlTag || !urlTag[1]) {
+        throw new Error("No URL found in response");
+    }
+
+    // If the URL is already absolute, use it as is
+    if (urlTag[1].startsWith("http://") || urlTag[1].startsWith("https://")) {
+        return urlTag[1];
+    }
+
+    // Combine the base download URL with the relative path
+    const cleanBaseUrl = baseDownloadUrl.replace(/\/$/, "");
+    const cleanPath = urlTag[1].replace(/^\//, "");
+    console.log(`${cleanBaseUrl}/${cleanPath}`);
+    return `${cleanBaseUrl}/${cleanPath}`;
+}
+
 async function handleImageUpload(event: Event, field: "picture" | "banner") {
     const input = event.target as HTMLInputElement;
     if (input.files?.[0]) {
         const file = input.files[0];
-        // TODO: Implement image upload to a hosting service
-        // For now, we'll just use a local URL
-        metadata[field] = URL.createObjectURL(file);
+        try {
+            // Set loading state
+            if (field === "banner") {
```

Comment:
Nit: Perhaps define

```ts
const isBanner = field === "banner";
```

and then check `isBanner` rather than making the string comparison each time